independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/ServerSerialisers.java

```diff
@@ -502,4 +519,15 @@ public static void encodeResponseHeaders(ResteasyReactiveRequestContext requestC
         }
     }
 
+    private static boolean useSet(String header) {
+        if (!header.startsWith(CONTENT)) {
+            return false;
+        }
+        if (header.length() < CONTENT.length() + 2) {
+            return false;
+        }
+        String substring = header.substring(CONTENT.length() + 1);
+        return substring.equals(TYPE) || substring.equals(LENGTH);
+    }
```

Comment:
Just a preference but this seems very complicated, where:
```java
private static boolean useSet(String header) {
    return header.startsWith(CONTENT_TYPE) || header.startsWith(CONTENT_LENGTH);
}
```
would have done it.

Also you sure the header will have this exact case? Are they not partly out of our control? Given headers are case insensitive, I think it's safer to do the following as you never know:
```java
private static boolean useSet(String header) {
    String lowercaseHeader = header.toLowerCase(Locale.ROOT);
    return lowercaseHeader.startsWith(CONTENT_TYPE_LOWERCASE) || lowercaseHeader.startsWith(CONTENT_LENGTH_LOWERCASE);
}
```
where the constants are lowercase versions of the header.

Reply:
> would have done it.

It would, but I wanted to make it as fast as possible.

Reply:
> Also you sure the header will have this exact case?

I'll do another check, but I am pretty sure that at this point, the case doesn't matter.

Reply:
PR updated